server/src/main/java/org/opensearch/index/translog/RemoteFsTimestampAwareTranslog.java

```diff
@@ -189,7 +189,7 @@ public void onResponse(List<BlobMetadata> blobMetadata) {
                     List<String> metadataFilesToBeDeleted = getMetadataFilesToBeDeleted(metadataFiles, indexDeleted);
 
                     // If index is not deleted, make sure to keep latest metadata file
-                    if (indexDeleted == false || RemoteStoreSettings.isPinnedTimestampsEnabled()) {
+                    if (indexDeleted == false) {
```

Comment:
(Nit) I personally would prefer `if (!indexDeleted)` but I understand other maintainers prefer this form.  It all compiles to the same code anyway.

Reply:
Yeah, this is mostly to keep the syntax consistent with rest of the code.